### PR TITLE
ctest test merged with regression_tests.tcl

### PR DESCRIPTION
**Pull request by gadfort**
Ensures all tests that appear in regression_tests.tcl are present in CMakeLists.txt for the tool test dir.

No functional changes.

Notes:
- Adds the base openroad tests as well, which were missing. I had to disable timing_api_4 as it was generating a failure on my machine.
- Only includes tests from record_tests list

Additionally: any new tests getting added after this PR must appear in both files until a followon PR removes the dependency on `./regression` in the tool directories

### CODE REVIEW

File: src/cts/test/CMakeLists.txt [hunk 1]
```diff
@@ -7,29 +7,37 @@
 include("openroad")
 
 set(TEST_NAMES
+    simple_test_hier
+    array
```
Comment (maliberty) [inline src/cts/test/CMakeLists.txt:11]: Any reason this isn't sorted?  Likewise lvt_lib

**Comment by maliberty** [comment]
Does timing_api_4 fail when run from the regular regression script?

File: src/dft/test/CMakeLists.txt [hunk 2]
```diff
@@ -2,17 +2,24 @@
...
+    scan_architect_clock_mix_sky130
+    scan_architect_register_bank_no_clock_mix_sky130
+    scandef_sky130
+    scandef_core_sky130
+    max_chain_count_sky130
 )
 
+# Skipped
+#dft_man_tcl_check
+#dft_readme_msgs_check
```
[the remark this answers is not in the document]
Comment (gadfort) [reply src/dft/test/CMakeLists.txt:19]: they are disabled in the tcl regression so I made a note of their exclusion.

File: src/cts/test/CMakeLists.txt (hunk 1 above)
Comment (gadfort) [reply src/cts/test/CMakeLists.txt:11]: order was kept from the record_tests list (but it would be trivial to sort them if you want)

**Comment by gadfort** [comment]
> Does timing_api_4 fail when run from the regular regression script?

Yes:
```
timing_api_4 (py) *ERROR* segmentation violation
------------------------------------------------------
Errored 1/1
```

**Comment by maliberty** [comment]
@gadfort please sort, merge master, and enable timing_api_4

**Comment by gadfort** [comment]
@maliberty sorted and merged

**Review by maliberty — APPROVED** [review]

**Pull request merged**